airbyte-ci/connectors/live-tests/src/live_tests/conftest.py

```diff
@@ -270,6 +272,14 @@ def get_option_or_fail(config: pytest.Config, option: str) -> str:
     pytest.fail(f"Missing required option: {option}")
 
 
+def get_control_version(config: pytest.Config) -> str:
+    if control_version := config.getoption("--control-version"):
+        return control_version
+    if connector_docker_repository := config.getoption("--connector-image"):
+        return get_current_docker_image_tag(connector_docker_repository)
```

Comment:
If we want open source users to be able to run the tests in the future, they won't have access to the connection-retriever. I understand that we have a lot of other work to do to make the live tests able to handle that scenario, but if possible it feels like we shouldn't introduce more places where we depend on it.

I think we could achieve that here by passing e.g. `latest` (or `None`) to the connection-retriever, and letting it call `get_current_docker_image_tag` on its own instead of calling it here.

Then to get the real control version displayed in the report (I don't think we want to display "latest") we may need to get it off of the `ConnectionObjects` object. Since we're already relying on `ConnectionObjects` elsewhere, that change wouldn't feel to me like we're introducing a new dependency on the connection-retriever.

Reply:
The `get_current_docker_image_tag` logic is just fetching the public cloud registry and getting the `dockerImageTag` value. I used it from the `connection_retriever` to avoid re-implementing the same function twice, but if we ever make the `connection_retriever` deps optional it'd be pretty straightfoward to copy it to a local util. I'd prefer to not use `latest` at all as it's not carrying any technical meaning: even if an image with the `latest` tag is available does not guarantee that this image is released (if publish fails halfway).

Reply:
That makes sense to me.